src/sink/gcs.rs

```diff
@@ -0,0 +1,345 @@
+                    file_path.to_string()
+                } else {
+                    return Err("Invalid Command, expected `file_path` field".into());
+                },
+            ),
+            "remove_object" => StorageCommand::Remove(
+                if let Some(Value::String(bucket_name)) = o.get("bucket") {
+                    bucket_name.to_string()
+                } else {
+                    return Err("Invalid Command, expected `bucket` field".into());
+                },
+                if let Some(Value::String(object_name)) = o.get("object") {
+                    object_name.to_string()
+                } else {
+                    return Err("Invalid Command, expected `object` field".into());
+                },
+            ),
+            "create_bucket" => StorageCommand::Create(
+                if let Some(Value::String(project_id)) = o.get("project_id") {
+                    project_id.to_string()
+                } else {
+                    return Err("Invalid Command, expected `project_id` field".into());
+                },
+                if let Some(Value::String(bucket_name)) = o.get("bucket") {
+                    bucket_name.to_string()
+                } else {
+                    return Err("Invalid Command, expected `bucket` field".into());
+                },
+            ),
+            "delete_bucket" => {
+                StorageCommand::Delete(if let Some(Value::String(bucket_name)) = o.get("bucket") {
+                    bucket_name.to_string()
+                } else {
+                    return Err("Invalid Command, expected `bucket` field".into());
+                })
+            }
+            "download_object" => StorageCommand::Download(
+                if let Some(Value::String(bucket_name)) = o.get("bucket") {
+                    bucket_name.to_string()
+                } else {
+                    return Err("Invalid Command, expected `bucket` field".into());
+                },
+                if let Some(Value::String(object_name)) = o.get("object") {
+                    object_name.to_string()
+                } else {
+                    return Err("Invalid Command, expected `object` field".into());
+                },
+            ),
+            _ => StorageCommand::Unknown,
+        };
+        return Ok(command);
+    }
+
+    Err("Invalid Command".into())
+}
+
+#[async_trait::async_trait]
+impl Sink for GoogleCloudStorage {
+    async fn terminate(&mut self) {}
+
+    async fn on_event(
+        &mut self,
+        _input: &str,
+        _codec: &mut dyn Codec,
+        _codec_map: &HashMap<String, Box<dyn Codec>>,
+        mut event: Event,
+    ) -> ResultVec {
+        let remote = if let Some(remote) = &self.remote {
+            remote
+        } else {
+            self.remote = Some(auth::json_api_client(&self.config.pem, &HeaderMap::new()).await?);
+            let remote = self.remote.as_ref().ok_or("Client error!")?;
```

Comment:
That's nice I dind't knew you could include a `&str` in the `ok_or` I always used Error::from() this is a nice trick. Thank you!